### ISSUE TITLE
Crash when attempting to use xla.conv with complex inputs

### ISSUE THREAD

**Issue by bchetioui** [tensorflow/tensorflow#43300]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Debian 10.4
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): v1.12.1-41557-gae0a324182 2.4.0-dev20200915
- Python version: Python 3.8.3
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: N/A
- GPU model and memory: N/A

**Describe the current behavior**

TF throws an exception when trying to use `tensorflow.compiler.tf2xla.python.xla.conv` with complex inputs, and the stack trace indicates it may be a bug in the HLO -> LLVM IR lowering.

**Standalone code to reproduce the issue**

```python
import numpy as np
from tensorflow.compiler.tf2xla.python import xla as tfxla
from tensorflow.compiler.xla import xla_data_pb2

proto = xla_data_pb2.ConvolutionDimensionNumbers()
proto.input_batch_dimension = 0
proto.input_feature_dimension = 1
proto.output_batch_dimension = 0
proto.output_feature_dimension = 1
proto.kernel_output_feature_dimension = 0
proto.kernel_input_feature_dimension = 1
proto.input_spatial_dimensions.extend([2, 3])
proto.kernel_spatial_dimensions.extend([2, 3])
proto.output_spatial_dimensions.extend([2, 3])

lhs = np.ones((2,3,9,10), dtype=np.complex64)
rhs = np.ones((3,3,4,5), dtype=np.complex64)

padding = ((0, 0), (0, 0))
window_strides, lhs_dilation, rhs_dilation = (1, 1), (1, 1), (1, 1)
feature_group_count = 1

tfxla.conv(lhs, rhs, window_strides, padding, lhs_dilation, rhs_dilation,
           proto, feature_group_count=feature_group_count, precision_config=None)
```

**Other info / logs**

Stack trace:

```
2020-09-17 16:13:20.385677: E tensorflow/compiler/xla/status_macros.cc:56] Internal: RET_CHECK failure (tensorflow/compiler/xla/service/cpu/cpu_compiler.cc:501) !llvm::verifyModule(llvm_module, &err_stream) Invalid LLVM IR before optimizations:
Floating-point arithmetic operators only work with floating-point types!
  %60 = fmul reassoc nsz contract %complex64 %57, %59
Floating-point arithmetic operators only work with floating-point types!
  %62 = fadd reassoc nsz contract %complex64 %61, %60

This probably indicates a bug in the HLO -> LLVM IR lowering. Rerun with --xla_dump_to to get the IR. 
*** Begin stack trace ***

        xla::status_macros::MakeErrorStream::Impl::GetStatus()

        xla::cpu::CpuCompiler::RunBackend(std::unique_ptr<xla::HloModule, std::default_delete<xla::HloModule> >, stream_executor::StreamExecutor*, stream_executor::DeviceMemoryAllocator*)
        xla::Service::BuildExecutable(xla::HloModuleProto const&, std::unique_ptr<xla::HloModuleConfig, std::default_delete<xla::HloModuleConfig> >, xla::Backend*, stream_executor::StreamExecutor*, stream_executor::DeviceMemoryAllocator*)
        xla::LocalService::CompileExecutables(xla::XlaComputation const&, absl::lts_2020_02_25::Span<xla::Shape const* const>, xla::ExecutableBuildOptions const&)
        xla::LocalClient::Compile(xla::XlaComputation const&, absl::lts_2020_02_25::Span<xla::Shape const* const>, xla::ExecutableBuildOptions const&)
        tensorflow::XlaCompilationCache::BuildExecutable(tensorflow::XlaCompiler::Options const&, tensorflow::XlaCompilationResult const&, std::unique_ptr<xla::LocalExecutable, std::default_delete<xla::LocalExecutable> >*)
        tensorflow::XlaCompilationCache::CompileImpl(tensorflow::XlaCompiler::Options const&, tensorflow::NameAttrList const&, absl::lts_2020_02_25::Span<tensorflow::XlaArgument const>, std::function<tensorflow::Status (tensorflow::XlaCompiler*, tensorflow::XlaCompilationResult*)> const&, absl::lts_2020_02_25::optional<long long>, tensorflow::XlaCompilationResult const**, xla::LocalExecutable**)
        tensorflow::XlaCompilationCache::CompileSingleOp(tensorflow::XlaCompiler::Options const&, absl::lts_2020_02_25::Span<tensorflow::XlaArgument const>, tensorflow::OpKernelContext*, tensorflow::XlaCompiler::CompileOptions const&, tensorflow::XlaCompilationResult const**, xla::LocalExecutable**)
        tensorflow::XlaCompileOnDemandOp::Compile(tensorflow::OpKernelContext*, tensorflow::XlaCompilationResult const**, tensorflow::XlaCompilationCache**, absl::lts_2020_02_25::flat_hash_map<int, absl::lts_2020_02_25::optional<tensorflow::Tensor>, absl::lts_2020_02_25::hash_internal::Hash<int>, std::equal_to<int>, std::allocator<std::pair<int const, absl::lts_2020_02_25::optional<tensorflow::Tensor> > > >*, xla::LocalExecutable**)
        tensorflow::XlaCompileOnDemandOp::Compute(tensorflow::OpKernelContext*)
        tensorflow::KernelAndDeviceOp::Run(tensorflow::ScopedStepContainer*, tensorflow::EagerKernelArgs const&, std::vector<absl::lts_2020_02_25::variant<tensorflow::Tensor, tensorflow::TensorShape>, std::allocator<absl::lts_2020_02_25::variant<tensorflow::Tensor, tensorflow::TensorShape> > >*, tensorflow::CancellationManager*, absl::lts_2020_02_25::optional<tensorflow::EagerRemoteFunctionParams> const&)
        tensorflow::EagerKernelExecute(tensorflow::EagerContext*, absl::lts_2020_02_25::InlinedVector<tensorflow::TensorHandle*, 4ul, std::allocator<tensorflow::TensorHandle*> > const&, absl::lts_2020_02_25::optional<tensorflow::EagerRemoteFunctionParams> const&, std::unique_ptr<tensorflow::KernelAndDevice, tensorflow::core::RefCountDeleter> const&, tensorflow::GraphCollector*, tensorflow::CancellationManager*, absl::lts_2020_02_25::Span<tensorflow::TensorHandle*>)
        tensorflow::ExecuteNode::Run()
        tensorflow::EagerExecutor::SyncExecute(tensorflow::EagerNode*)

        tensorflow::EagerExecute(tensorflow::EagerOperation*, tensorflow::TensorHandle**, int*)
        tensorflow::EagerOperation::Execute(absl::lts_2020_02_25::Span<tensorflow::AbstractTensorHandle*>, int*)
        TFE_Execute
        TFE_Py_ExecuteCancelable(TFE_Context*, char const*, char const*, absl::lts_2020_02_25::InlinedVector<TFE_TensorHandle*, 4ul, std::allocator<TFE_TensorHandle*> >*, _object*, TFE_CancellationManager*, absl::lts_2020_02_25::InlinedVector<TFE_TensorHandle*, 2ul, std::allocator<TFE_TensorHandle*> >*, TF_Status*)




        _PyObject_MakeTpCall
        _PyEval_EvalFrameDefault
        _PyEval_EvalCodeWithName
        _PyFunction_Vectorcall
        _PyEval_EvalFrameDefault
        _PyEval_EvalCodeWithName
        _PyFunction_Vectorcall
        _PyEval_EvalFrameDefault
        _PyEval_EvalCodeWithName
        _PyFunction_Vectorcall
        _PyEval_EvalFrameDefault
        _PyEval_EvalCodeWithName
        _PyFunction_Vectorcall
        _PyEval_EvalFrameDefault
        _PyEval_EvalCodeWithName
        PyEval_EvalCode

        PyRun_FileExFlags
        PyRun_SimpleFileExFlags

        Py_BytesMain
        __libc_start_main
        _start
*** End stack trace ***
```

**Comment by bhack** [comment]
If I remember correctly `xla.conv` doesn't support complex type inputs. 
See e.g. the recent Jax patch:
https://github.com/google/jax/pull/3735

**Comment by ravikyram** [comment]
I have tried in colab with TF versions 2.3, nightly versions(`2.4.0-dev20200917`) and was able to reproduce the issue.Please, find the gist [here](https://colab.research.google.com/gist/ravikyram/0ea6e759643cd4078d5caf742db8c1b8/untitled379.ipynb). Thanks!

**Comment by Saduf2019** [comment]
@bchetioui 
This has been fixed on tf 2.5, please refer to the [gist here](https://colab.research.google.com/gist/Saduf2019/37b3b34cf9b4539c628a6f164c2cb018/untitled597.ipynb), and move this to closed status.

**Comment by bchetioui** [comment]
Thank you for the update! I have run it on my machine as well and confirmed it now works.
I am thus closing the issue.
